src/python/WMCore/Services/Rucio/Rucio.py

```diff
@@ -757,18 +757,21 @@ def pickRSE(self, rseExpression='rse_type=TAPE\cms_type=test', rseAttribute='ddm
         rsesWeight = []
 
         for rse in matchingRSEs:
-            attrs = self.cli.list_rse_attributes(rse)
+            rseAttrs = self.cli.list_rse_attributes(rse)
             if rseAttribute:
                 try:
-                    quota = float(attrs.get(rseAttribute, 0))
+                    attrValue = float(rseAttrs.get(rseAttribute, 0))
                 except (TypeError, KeyError):
-                    quota = 0
+                    attrValue = 0
             else:
-                quota = 1
-            requiresApproval = attrs.get('requires_approval', False)
-            if quota > minNeeded:
+                attrValue = 1
+            requiresApproval = rseAttrs.get('requires_approval', False)
+            if rseAttribute == "ddm_quota" and attrValue > minNeeded:
+                rsesWithApproval.append((rse, requiresApproval))
+                rsesWeight.append(attrValue)
+            elif rseAttribute != "ddm_quota":  # e.g. dm_weight
```

Comment:
is `dm_weight` the only possible `rseAttribute` value different from ddm_quota

Reply:
Nope, there are a few. Here is an example output from my tests yesterday:
```
{'T1_IT_CNAF_Tape': True,
 'archive_timeout': '605000',
 'cms_type': 'real',
 'country': 'IT',
 'ddm_quota': '2368393765836074',
 'dm_weight': '22',
 'fts': 'https://***',
 'greedyDeletion': False,
 'lfn2pfn_algorithm': 'cmstfc',
 'loadtest': True,
 'pnn': 'T1_IT_CNAF_Buffer',
 'quota_approvers': '***,
 'reaper': False,
 'requires_approval': False,
 'rule_approvers': '**',
 'site_admins': '***',
 'tier': '1',
 'update_from_json': True,
 'wmcore_output_tape': True}
```

Reply:
Ok, so we are implicitly assuming that this function works either with `rseAttribute='ddm_quota'` or `rseAttribute='dm_weight'`, right? If yes, would it be better to have
```
elif rseAttribute != "dm_weight":
     ...
else:
    continue
```
for the sake of clarity?

For the rest, everything looks fine to me

Reply:
Yes, it is supposed to work for both attributes (ddm_quota and dm_weight).

The problem with your suggestion is that it would accept a "ddm_quota" attribute that does not meet the expected quota. While the expected behavior is: if it is "ddm_quota", then we need to check that there is enough quota for the dataset size to be transferred.

The implementation I have in this PR is generic enough IMO. It works with "ddm_quota" (+ the actual quota) and it also works with any other RSE attribute that we decide to pass over and which is a number.

Reply:
I think your last comment solves my doubts, thanks!